aws/request/handlers.go

```diff
@@ -20,22 +20,6 @@ type Handlers struct {
 	AfterRetry       HandlerList
 }
 
-// Copy returns of this handler's lists.
-func (h *Handlers) Copy() Handlers {
```

Comment:
These copy's are actually needed. `HandlerList` is a struct, but it contains a slice of `Handlers`.  This slice needs to be independently mutable at each stage of a request's pipeline without impacting any upstream functionality. 

Specifically this change will cause any modifications made to a request handler's list to be propagated upstream to the service's request handler list. This means that If a customization needs to modify a request handler to perform some functionality for a specific API operation, the mutation of the handler list will be propagated upstream to the service, and now all API operation made will now also include the request handler customization. A common pattern is to share `session.Session` between multiple service clients, and this would polute all other service client instances with the customized request handlers.

This also introduces race conditions in SDK. Which is the panic shown in the failed Travis test. Since the backing arrays of handlers would be shared between multiple instances of service clients, and API operations being made concurrently, any mutation to the handler list would introduce race conditions.

This [Go blog post](https://blog.golang.org/slices) provides a good rundown on how Arrays vs Slices could break if down stream changes make unintended mutations to a upstream owned slice.